pymaker/deployment.py

```diff
@@ -231,120 +243,49 @@ def __init__(self, web3: Web3, config: Config):
-
-        config = DssDeployment.Config(mom, vat, vow, drip, pit, cat, flap, flop, dai, dai_adapter, dai_move, mkr)
-        deployment = DssDeployment(web3, config)
-
-        collateral = Collateral.deploy(web3=web3, name='WETH', vat=vat)
-        deployment.deploy_collateral(collateral,
-                                     debt_ceiling=Wad.from_number(100000),
-                                     penalty=Ray.from_number(1),
-                                     flop_lot=Wad.from_number(10000),
-                                     ratio=Ray.from_number(1.5),
-                                     initial_price=Wad.from_number(219))
-
-        return deployment
-
-    def deploy_collateral(self, collateral: Collateral,
-                          debt_ceiling: Wad, penalty: Ray, flop_lot: Wad, ratio: Ray, initial_price: Wad):
-
-        collateral.pip = DSValue.deploy(web3=self.web3)
-        assert collateral.pip.poke_with_int(initial_price.value).transact()  # Initial price
-
-        collateral.flipper = Flipper.deploy(web3=self.web3, dai=self.dai_move.address, gem=collateral.mover.address)
-
-        collateral.spotter = Spotter.deploy(web3=self.web3, pit=self.pit.address, ilk=collateral.ilk)
-        collateral.spotter.file_pip(collateral.pip.address).transact()
-        collateral.spotter.file_mat(ratio).transact()  # Liquidation ratio
-
-        assert collateral.gem.approve(collateral.adapter.address).transact()
-
-        assert self.cat.file_flip(collateral.ilk, collateral.flipper).transact()
-        assert self.cat.file_lump(collateral.ilk, flop_lot).transact()  # Liquidation Quantity
-        assert self.cat.file_chop(collateral.ilk, penalty).transact()  # Liquidation Penalty
-        assert self.vat.init(collateral.ilk).transact()
-        assert self.drip.init(collateral.ilk).transact()
-
-        assert self.vat.rely(collateral.flipper.address).transact()
-        assert self.vat.rely(collateral.adapter.address).transact()
-        assert self.vat.rely(collateral.mover.address).transact()
-        assert self.pit.rely(collateral.spotter.address).transact()
-
-        assert self.pit.file_line(collateral.ilk, debt_ceiling).transact()  # Collateral debt Ceiling
-        assert collateral.spotter.poke().transact()
-
-        self.collaterals.append(collateral)
+    def approve_dai(self, usr: Address):
+        """
+        Allows the user to draw Dai from and repay Dai to their CDPs.
+
+        Args
+            usr: Recipient of Dai from one or more CDPs
+        """
+        assert isinstance(usr, Address)
+
+        self.dai_adapter.approve(approval_function=hope_directly(), source=self.vat.address,
+                                 from_address=usr)
+        self.dai.approve(self.dai_adapter.address).transact()
+
+    def active_auctions(self) -> dict:
+        flips = {}
+        for collateral in self.collaterals.values():
+            flips[collateral.ilk.name] = collateral.flipper.active_auctions()
```

Comment:
I am not sure I understand scope of creating dict with same filp auction for all collaterals in use. can you pls explain